src/main/java/org/jitsi/jigasi/xmpp/ChatRoomMemberPresenceExtensionReader.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Jigasi, the JItsi GAteway to SIP.
+ *
+ * Copyright @ 2018 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.jigasi.xmpp;
+
+import net.java.sip.communicator.impl.protocol.jabber.*;
+import net.java.sip.communicator.impl.protocol.jabber.extensions.jitsimeet.*;
+import net.java.sip.communicator.service.protocol.*;
+import org.jitsi.jigasi.util.*;
+import org.jitsi.util.*;
+import org.jivesoftware.smack.packet.*;
+
+/**
+ * This class is able to receive {@link Presence} objects and update data
```

Comment:
We read the identity fields from the `ChatRoomMemberJabberImpl` presence and store them in the `ChatRoomMemberJabberImpl` in another format (using the `DataObject`) API.

Why don't we just define helper functions to read this data from presence?

Reply:
the lastPresence might be overwritten, so we need to either store all presences retrieved over the lifetime of the object, implement member fields in `ChatRoomMemberJabberImpl` with getters/setters, or use the DataObject API